components/calendly/sources/common-webhook.js

```diff
@@ -0,0 +1,52 @@
+const calendly = require("../calendly.app.js");
+const get = require("lodash/get");
+
+module.exports = {
+  props: {
+    calendly,
+    db: "$.service.db",
+    http: {
+      type: "$.interface.http",
+      customResponse: true,
+    },
+  },
+  hooks: {
+    async activate() {
+      const events = this.getEvents();
+      const body = {
+        url: this.http.endpoint,
+        events,
+      };
+      const resp = await this.calendly.createHook(body);
+      this.db.set("hookId", resp.data.id);
+    },
+    async deactivate() {
+      await this.calendly.deleteHook(this.db.get("hookId"));
+    },
+  },
+  methods: {
+    generateMeta(body) {
+      const eventId = get(body, "payload.event.uuid");
```

Comment:
Are these properties guaranteed to exist for all events? I see the intent behind this change, so that you can use this as a default for the invitee sources and override `generateMeta` only as necessary, but this might yield an empty string for the `id` if another component _doesn't_ override it and `payload.invitee.uuid` (for example) isn't defined.

Even though it's a little more code, I'd recommend having the default `generateMeta` throw a similar error we do elsewhere to warn authors if they fail to implement it:

```javascript
throw new Error("generateMeta is not implemented");
```

Then, you can define an additional method (e.g. `generateInviteeMeta`) here in the common file, and have `generateMeta` in the invitee-related components reference that:

```javascript
methods: {
  ...common.methods,
  generateMeta(body) {
    return this.generateInviteeMeta(body)
  }
}
```

Would that work?